app/models/deletion.rb

```diff
@@ -7,12 +7,18 @@ class Deletion < ActiveRecord::Base
 
   before_validation :record_metadata
   after_create :remove_from_index, :set_yanked_info_checksum
-  after_commit :remove_from_storage
+  after_commit :remove_from_storage, on: :create
   after_commit :expire_cache
   after_commit :update_search_index
 
   attr_accessor :version
 
+  def restore!
+    restore_to_index
+    restore_to_storage
+    destroy!
```

Comment:
Given that we want this restored version to be yankable in future, adding a `restored` flag would introduce two paths for deletion/yanking of a version. One where we create a new `deletion` record and other where we check if `deletion` record already exists and mark `restored = false`. I think it would create unnecessary confusion.
Can you please explain what is the value of marking it restored instead of deleting the record?